### PR TITLE
GlobalCollect: Support Airline Data

### PR DESCRIPTION
**Pull request by naashton**
Expand support for a broader set of Airline data with Global Collect.

CE-2052

Unit: 30 tests, 146 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Remote: 25 tests, 60 assertions, 2 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
92% passed

### CODE REVIEW

**Review by dsmcclain — DISMISSED** [review]
The two failing tests I get are:
`test_successful_purchase_with_installments`
`test_successful_purchase_with_airline_data`

The installments test is failing on master as well, so I'm not really worried about that at the moment.

The airline_data test fails with a timeout, like you described. Have you reached out to the gateway about this? It could just be a limitation of their sandbox, but considering that these are the exact fields that this PR is addressing it would nice to at least have an answer from them about why the airline data might not be working.

**Comment by naashton** [comment]
Erik is linking me up with Igenico's POC to see why we are getting a time-out. I'll re-request review once we are able to get something to work with.

File: lib/active_merchant/billing/gateways/global_collect.rb [hunk 1]
```diff
@@ -103,37 +103,71 @@ def add_order(post, money, options, capture: false)
...
-        airline_data['name'] = name if name
+        post['order']['additionalInput']['airlineData'] = airline_data
+      end
 
+      def add_flight_legs(airline_options)
         flight_legs = []
-        options[:airline_data][:flight_legs]&.each do |fl|
+        airline_options[:flight_legs]&.each do |fl|
           leg = {}
+          leg['airlineClass'] = fl[:carrier_code] if fl[:airline_class]
```
Comment (jessiagee) [inline lib/active_merchant/billing/gateways/global_collect.rb:136]: Should this be `fl[:airline_class] if fl[:airline_class]` ?

File: lib/active_merchant/billing/gateways/global_collect.rb [hunk 2]
```diff
@@ -103,37 +103,71 @@ def add_order(post, money, options, capture: false)
...
           leg['arrivalAirport'] = fl[:arrival_airport] if fl[:arrival_airport]
-          leg['originAirport'] = fl[:origin_airport] if fl[:origin_airport]
+          leg['arrivalTime'] = fl[:arrival_time] if fl[:arrival_time]
+          leg['carrierCode'] = fl[:carrier_code] if fl[:carrier_code]
+          leg['conjunctionTicket'] = fl[:conjuction_ticket] if fl[:conjction_ticket]
```
Comment (jessiagee) [inline lib/active_merchant/billing/gateways/global_collect.rb:140]: Typos - `fl[:conjunction_ticket] if fl[:conjunction_ticket]`

**Review by jessiagee — CHANGES_REQUESTED** [review]
Some typos to clean up

**Review by jessiagee — APPROVED** [review]
bundle exec rake test:local

5010 tests, 74788 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Running RuboCop...

725 files inspected, no offenses detected

Loaded suite test/unit/gateways/global_collect_test

31 tests, 154 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

Loaded suite test/remote/gateways/remote_global_collect_test

Failure: test_successful_purchase_with_installments(RemoteGlobalCollectTest) -- failing on master as well

26 tests, 62 assertions, 1 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
96.1538% passed

**Pull request merged**